Tools/TestWebKitAPI/Tests/WTF/UUID.cpp

```diff
@@ -95,3 +95,11 @@ TEST(WTF, TestUUIDVersion4MakeString)
 
     EXPECT_EQ(WTF::StringTypeAdapter<UUID>(uuid.value()).length(), 36U);
 }
+
+TEST(WTF, TestUUIDDeletedValue)
+{
+    String deletedValueString = "00000000-0000-0000-0000-000000000001"_s;
+    auto uuid = UUID::parse(deletedValueString);
+    EXPECT_TRUE(!!uuid);
+    EXPECT_FALSE(uuid->isValid());
```

Comment:
By default, parse should return a valid value or std::null opt.
It might be fine to make parse() take a AllowInvalidValue boolean enum parameter (with a default value equal to No to not change the default behavior).
Only UID::UUID(NSUUID *nsUUID) would then call parse() with AllowDeletedValue::Yes.